scheduler/src/main/java/gov/nasa/jpl/aerie/scheduler/goals/RecurrenceGoal.java

```diff
@@ -178,10 +178,12 @@ private java.util.Collection<MissingActivityConflict> makeRecurrenceConflicts(
       //        framed, which is exactly what the current solver will do since
       //        it chooses the minimum. but it looks ugly. so for now passing
       //        a limited flexibility for creation
-      var interval =  new Windows(Window.between(intervalT.minus(recurrenceInterval.end), Duration.min(intervalT, end)));
-      conflicts.add(new MissingActivityTemplateConflict(
-          this, new Windows(
-          interval),this.getActTemplate()));
+      var interval = Window.between(intervalT.minus(recurrenceInterval.end), Duration.min(intervalT, end));
+      if(!interval.isSingleton()) {
```

Comment:
I'm curious why this is a problem - if an activity has zero duration, it can fit into a singleton interval, right?

Reply:
This has more to do with the semantics of the goal. This is a corner case happening when the goal planning horizon is  a multiple of the recurrence interval. Ex: 8h horizon for a 1hr recurrence. We go through each hour and find that we need one activity for each. At some point, we have `intervalT` = 9, we do i`ntervalT.minus(1)` = 8, we get `intervalT`=8. For the other part of the resulting interval, we take the minimum between end of planning horizon (to avoid setting the domain of the conflict outside of the overall horizon), here 8, and `intervalT` = 8. Which makes an empty interval. But actually, as the interval is empty, it means we do not need to schedule anything there, we already satisfied the goal of 1 act every hour. 
If the end of the scheduling horizon was at 8h05, we would need an activity between 8h and 8h05, and that's effectively what it would schedule.

Reply:
I'm a little confused myself. The endpoints of a `Window` can be either inclusive or exclusive; if you're treating a singleton as though it were empty, it suggests maybe the endpoints should be exclusive rather than inclusive, so that the window *is* actually empty.

Reply:
After discussing with @mattdailis, I have reverted the change and only replaced the break statement from `if(intervalT.compareTo(end) > 0)` by `if(intervalT.compareTo(end) >= 0)`, making `end` an exclusive bound. 
This procedure is called twice (above), once in the case where there is too large of a gap between 2 activities, then we are sure we do not want to include the `end` as it corresponds to the start of the second activity. For the other call, `end` corresponds to the end of the `temporalContext` which is the scheduling horizon of the goal.